infra/modules/waf/main.tf

```diff
@@ -0,0 +1,375 @@
+    key_arn  = aws_kms_key.waf_logging.arn
+  }
+  extended_s3_configuration {
+    role_arn   = aws_iam_role.firehose_perms.arn
+    bucket_arn = aws_s3_bucket.waf_logging.arn
+  }
+}
+
+# IAM Role for Kinesis
+resource "aws_iam_role" "firehose_perms" {
+  name               = "placeholder"
+  description        = "IAM role for the KDF"
+  assume_role_policy = data.aws_iam_policy_document.firehose_assume_role.json
+}
+
+# assume role
+data "aws_iam_policy_document" "firehose_assume_role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["firehose.amazonaws.com"]
+    }
+  }
+}
+
+# role policy
+data "aws_iam_policy_document" "firehose_perms" {
+  statement {
+    sid    = "AccessKDF"
+    effect = "Allow"
+    actions = [
+      "kinesis:Get*",
+      "kinesis:PutRecord",
+      "s3:GetBucket",
+      "s3:PutObject"
+    ]
+    resources = [aws_s3_bucket.waf_logging.arn, "${aws_s3_bucket.waf_logging.arn}/*"]
+  }
+}
+
+# KMS key config
+resource "aws_kms_key" "waf_logging" {
+  enable_key_rotation = true
+  description         = "KMS key for encrypting WAF logs"
+}
+# S3 to store logging resources
+resource "aws_s3_bucket" "waf_logging" {
+  # checkov:skip=CKV2_AWS_62:Disable SNS requirement
+  # checkov:skip=CKV_AWS_144:Cross region replication not required by default
+  bucket = "${var.waf_name}-logs"
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
+resource "aws_s3_bucket_versioning" "waf_logging" {
+  bucket = aws_s3_bucket.waf_logging.id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_public_access_block" "waf_logging" {
+  bucket = aws_s3_bucket.waf_logging.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "waf_logging" {
+  bucket = aws_s3_bucket.waf_logging.id
+  rule {
+    apply_server_side_encryption_by_default {
+      kms_master_key_id = aws_kms_key.waf_logging.arn
+      sse_algorithm     = "aws:kms"
+    }
+    bucket_key_enabled = true
+  }
+}
+
+# give permission to upload into logging bucket
+resource "aws_s3_bucket_policy" "waf_logging" {
+  bucket = aws_s3_bucket.waf_logging.id
+  policy = data.aws_iam_policy_document.waf_logging_bucket.json
+}
+
+data "aws_iam_policy_document" "waf_logging_bucket" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:AbortMultipartUpload",
+      "s3:GetBucketLocation",
+      "s3:GetObject",
+      "s3:ListBucket",
+      "s3:ListBucketMultipartUploads",
+      "s3:PutObject"
+    ]
+    resources = [aws_s3_bucket.waf_logging.arn, "${aws_s3_bucket.waf_logging.arn}/*"]
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_caller_identity.current.account_id]
+    }
+  }
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "waf_logging" {
```

Comment:
I think you can call the s3-encrypted module for all of this s3 config and just pass in some variables. We don't yet have the lifecycle configuration in the s3-encrypted module, so you could move this one resource configuration block into that module, set some defaults (probably whatever makes most sense for the doc upload), and add some variables for any necessary overriding of defaults for the WAF